DESIGN.md

```diff
@@ -0,0 +1,107 @@
+
+We also prefer not to rely on centralised services when providing users and devices with the needed authority. When a user starts using a new device, their existing device should be authorise the new one without any intervention from a central service. This can be done by giving the new device one or more tickets that can be verified on the spot.
+
+Rather than putting everything in a closed system, ODD puts everything out in the open. That means encryption is used to keep data private. These encryption keys and authorisation tickets give us a chance to escape from passwords, so preferably passwords are avoided.
+
+The ODD SDK relies on open protocols, this includes the [WNFS protocol](https://github.com/wnfs-wg) for the data management, and [UCAN](https://github.com/ucan-wg) to provide users and devices with the necessary authorisation.
+
+## Implementation
+
+ts-odd is written in Typescript, a Javascript dialect. The resulting code should be usable in any modern Javascript environment that supports the [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Crypto_API) and [Web Assembly](https://webassembly.org/). It uses the Rust implementations of [WNFS](https://github.com/wnfs-wg/rs-wnfs) and [UCAN](https://github.com/ucan-wg/rs-ucan) which compile to Web Assembly.
+
+Various other pieces are layered on top of these core protocols and assumptions, these are called "components" in ts-odd. These allow us to adapt to various technologies and environments. We'll start with the components that are closest to the data and go from there. These are:
+
+1. Depot
+2. Account
+3. Identifier & Agent
+4. Authority
+5. Channel
+6. DNS
+7. Storage
+8. Manners
+
+The following graph shows a rough overview how the components relate:
+
+```mermaid
+flowchart LR;
+    Identifier --> Account --> Agent --> AS(((Account service))):::service
+
+    FS[(File System)]:::subject --> Depot
+    FS[(File System)]:::subject --> Manners
+    FS[(File System)]:::subject --> DNS
+
+    Authority --> Channel
+
+    Storage --> Authority
+    Depot <--> Storage
+    Account <--> Storage
+    Agent --> Storage
+
+
+    classDef component fill:#e3d2c4,stroke:#caaa90
+    classDef service fill:#e7d0ea,stroke:#cc9bd2
+    classDef subject fill:#f3dee0,stroke:#dfa6ab
+
+    class Account,Agent,Authority,Channel,Depot,DNS,Identifier,Manners,Storage component
+```
+
+These are only some of the relationships, but it should you a rough idea of some important concepts in the SDK. Starting from the top we have the identifier component that provides an authorisation ticket to the account component, which in turn makes another ticket and passes it to some external account service. Note that this is just an example, it may not use an external service.
```

Comment:
Maybe highlight the component names in here to make them easier for the user to see at a glance? Also, capitalising them will make them consistent with the components in the paragraphs that follow

```suggestion
These are only some of the relationships, but they should give you a rough idea of some important concepts in the SDK. Starting from the top, we have the `Identifier` component that provides an authorisation ticket to the `Account` component, which in turn makes another ticket and passes it to some external account service. Note that this is just an example, it may not use an external service.
```